### ISSUE TITLE
Docker-compose is reading the default .env file from the docker-compose.yml folder and not the working directory

### ISSUE THREAD

**Issue by bechhansen** [docker/compose#8208]
## Description of the issue
When .env file and docker-compose.yml files are located in different folders, docker compose reads the file form the folder of the yml file. Previously it did, and according to documentation it should, read the .env file from the working folder.

## Context information (for bug reports)

```
Output of "docker-compose version"
docker-compose version 1.28.5, build c4eb3a1f
docker-py version: 4.4.4
CPython version: 3.9.0
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
```

```
Output of "docker version"
Client: Docker Engine - Community
 Cloud integration: 1.0.9
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:14:53 2021
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:15:47 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```

## Steps to reproduce the issue

1. Use docker compose by specifying a *.yml file in another folder, using variables
2. Add .env files to working directory containing variables used in the *.yml file
3. See that variables are not resolved for the services of the *.yml file

### Observed result
If an .env file is located in the folder of the *.yml, variables from this files is used

### Expected result
I expect that docker-compose would resolve variables from an .env file located in the working folder.

## Additional information

OS: Windows 10

**Comment by stingalleman** [comment]
I'm also having this issue. Seems like a really big bug?

**Comment by javierguzman** [comment]
Isn't this related to the project-directory option? Docker-compose 1.28 has change its behaviour in regards to what you are mentionining as far as I have seen.  I think you can take a look at that option, it might help you. 

In my case the problem is that shell environment variables are not resolved within the yml file.

**Comment by aurelijusrozenas** [comment]
Today I also stepped onto this bug on linux. Works fine before 1.28 version.
Steps to reproduce:
- start docker image to mess around (sorry, random image, but it has everything for testing):
```
docker run --rm -it -v /var/run/docker.sock:/var/run/docker.sock -uroot jetbrains/teamcity-agent:2020.2.2 bash
```
- execute inside docker bash. Creates test docker-compose project
```
# version 1.27.4 - no warnings everything's as expected
curl -L "https://github.com/docker/compose/releases/download/1.27.4/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
chmod +x /usr/local/bin/docker-compose
docker-compose --version
cd /tmp
mkdir subdir
cat <<EOT >> subdir/docker-compose.yaml
version: '3'
services:
    hello-world:
        image: hello-world
        labels:
            test: \$TEST
EOT
echo TEST=testas > .env
docker-compose -f subdir/docker-compose.yaml up

# version 1.28.0 - show warning "WARNING: The TEST variable is not set. Defaulting to a blank string."
curl -L "https://github.com/docker/compose/releases/download/1.28.0/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
chmod +x /usr/local/bin/docker-compose
docker-compose --version
docker-compose -f subdir/docker-compose.yaml up
```

**Comment by wungad** [comment]
This is intended and was introduced with 1.28:

> Docker Compose versions earlier than 1.28, load the .env file from the current working directory, where the command is executed, or from the project directory if this is explicitly set with the --project-directory option. This inconsistency has been addressed starting with +v1.28 by limiting the default .env file path to the project directory.

Source: https://docs.docker.com/compose/env-file/
